### PR TITLE
Transaction Card

### PR DESCRIPTION
**Pull request by pixelflips**
## Description
https://kajabi.atlassian.net/browse/MAN-970

Adds new transaction card to the system.

### Screenshots
Screen Shot 2021-02-04 at 11 59 36 AM

### Steps for testing
1. Visit http://localhost:4000/pages/breakout/object/transaction_card
2. Verify UI and docs
3. Verify Name and Product name truncate when viewed in smaller containers/screens

--

1. Visit http://localhost:6006/?path=/story/sage-transactioncard--default
2. Verify UI and props
3. Verify Name and Product name truncate when viewed in smaller containers/screens

## Related
- NA

### CODE REVIEW

File: docs/lib/sage_rails/app/views/sage_components/_sage_transaction_card.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<% dropdown_items = [
+  {
```
Comment (philschanely) [inline docs/lib/sage_rails/app/views/sage_components/_sage_transaction_card.html.erb:2]: Should this include some customization options or hooks of some sort for functionality?

File: packages/sage-react/lib/TransactionCard/configs.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const LABEL_COLORS = {
+  PUBLISHED: 'published',
+  DRAFT: 'draft',
+  DANGER: 'danger',
+  INFO: 'info',
+  WARNING: 'warning',
+  LOCKED: 'locked',
+};
+
+export const DROPDOWN_ITEMS = [
```
Comment (philschanely) [inline packages/sage-react/lib/TransactionCard/configs.js:16]: Same here... seems we may need this to have some hooks for functionality.

**Review by philschanely — APPROVED** [review]
This is superbly done, @pixelflips . Only thing remaining to consider is how to set up the dropdown in context. LMK if I can assist.

**Comment by pixelflips** [comment]
@philschanely I've adjusted the dropdown as we discussed. Would you mind doing one more pass and then I think we're good to go. Thanks in advance!

File: docs/app/views/examples/objects/transaction_card/_preview.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<%= sage_component SageTransactionCard, {
+  label_color: "published",
+  label_text: "New",
+  name: "Lilly Jones",
+  amount: "+ 450.00",
+  amount_color: "sage",
+  product_name: "Total Product Blueprint",
+  transaction_time: "10:32pm",
+  dropdown_options: [{
```
Comment (philschanely) [inline docs/app/views/examples/objects/transaction_card/_preview.html.erb:9]: 😍

**Review by philschanely — APPROVED** [review]
Fabulous! Well done! 🙌

**Pull request merged**